weave/trace_server/file_storage_credentials.py

```diff
@@ -0,0 +1,115 @@
+import base64
+from typing import Optional, TypedDict, Union
+
+from azure.core.credentials import TokenCredential
+from google.oauth2.credentials import Credentials as GCPCredentials
+
+from weave.trace_server import environment
+
+
+class AWSCredentials(TypedDict, total=False):
```

Comment:
I don't think this is correct.  I would have expected:
```py
class AWSCredentials(TypedDict):
    access_key_id: str
    secret_access_key: str
    session_token: NotRequired[str]
```

When you set `total=False`, it means there can be other keys here which would be wrong

Reply:
changed